test_suite/conftest.py

```diff
@@ -33,6 +35,20 @@ def check_markers(host, request):
     host_version = host.system_info.release
     distro_version = f'{host_distro}{host_version}'
 
+    # Skip the test if the distro is explicitely excluded
+    if exclude_on_marker:
+        exclude_on_marker_list = exclude_on_marker.args[0]
+        if host_distro in exclude_on_marker_list \
+                or distro_version in exclude_on_marker_list:
+            pytest.skip(f"This test doesn't apply to {distro_version}")
+
+    # If there is no run_on_marker and distro is not excluded execute supported
+    # tests
+    if not run_on_marker:
+        if distro_version in supported_distros_and_versions:
```

Comment:
I think this `if` is similar to the one in line 61. Do you think we can move the validation here instead?

Reply:
I don't think so or maybe I just don't see it. This checks executes the test if there is no run_on marker, the distro is supported and it wasn't filtered out by exclude. Following lines then process the run_on marker as the assumption here is that either run_on, exclude_on or both may be specified.

Reply:
Yeah I was thinking about combining both ifs since the result is similar, but that may add more nesting or complexity.